src/main/java/org/broadinstitute/hellbender/tools/exome/ExonCollection.java

```diff
@@ -199,7 +199,7 @@ default long exomeSize() {
      * @return never {@code null}.
      * @throws IllegalArgumentException if {@code exon} is {@code null}.
      */
-    SimpleInterval location(final E exon);
+    SimpleInterval location(final Object exon);
```

Comment:
Why change the parameter type from `E` to `Object`? It seems to me that the code in `ExonCollections` could be improved a lot if these parameters were strongly typed.